From 8da2f1f817d9fba0a023f19410c4ddff7c844802 Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Mon, 17 Jun 2024 13:07:51 +0300
Subject: [PATCH] [SW-188948] gaudi3: tpm: reduce wait time for status

This patch reduces the TPM delays from 5 ms to 200us.
This done to avoid long waits which contribute to
overall increase in boot time of gaudi3. This change
brings down the combined time to perform TPM startup
and TPM PCR extend by 60ms.

Change-Id: Ie5e513a4d63c54362ad8a1a9c3a7296a1ef83ae3
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/lib/common/tpm2.c b/lib/common/tpm2.c
index dcdcbb3..60fe3d0 100644
--- a/lib/common/tpm2.c
+++ b/lib/common/tpm2.c
@@ -315,7 +315,7 @@
 	int ret;
 
 	do {
-		mdelay(TPM_TIMEOUT_MS);
+		udelay(SLEEP_DURATION_LONG_US);
 		ret = tpm_spi_status(status);
 		if (ret)
 			return ret;
@@ -345,7 +345,7 @@
 		if (burstcount)
 			return burstcount;
 
-		mdelay(TPM_TIMEOUT_MS);
+		udelay(SLEEP_DURATION_LONG_US);
 	} while (!timeout_expired(start, MS_TO_US(stop)));
 
 	return -1;
@@ -488,7 +488,7 @@
 			return ret;
 		}
 
-		mdelay(TPM_TIMEOUT_MS);
+		udelay(SLEEP_DURATION_LONG_US);
 	} while (!timeout_expired(start, MS_TO_US(stop)));
 
 	preboot_puts("TPM2: Timeout getting locality\n");
@@ -643,7 +643,7 @@
 			return ret;
 		}
 
-		mdelay(TPM_TIMEOUT_MS);
+		udelay(SLEEP_DURATION_LONG_US);
 		if (timeout_expired(start, MS_TO_US(stop))) {
 			preboot_puts("TPM2: timeout expired\n");
 			ret = -1;
@@ -990,7 +990,7 @@
 	start = timestamp_get_us();
 	stop = TIS_LONG_TIMEOUT_MS;
 	do {
-		mdelay(TPM_TIMEOUT_MS);
+		udelay(SLEEP_DURATION_LONG_US);
 
 		ret = tpm_tis_spi_read_silent(TPM_ACCESS(loc), &status, 1);
 		if (ret == -ETIMEDOUT)
